muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/InstrumentationModuleMuzzle.java

```diff
@@ -22,11 +24,35 @@
    */
   Map<String, ClassRef> getMuzzleReferences();
 
+  /** See {@link #getMuzzleReferences()} */
+  static Map<String, ClassRef> getMuzzleReferences(InstrumentationModule module) {
+    if (module instanceof InstrumentationModuleMuzzle) {
+      return ((InstrumentationModuleMuzzle) module).getMuzzleReferences();
+    } else {
+      return Collections.emptyMap();
+    }
+  }
+
   /**
    * Builds the associations between instrumented library classes and instrumentation context
    * classes. Keys (and their subclasses) will be associated with a context class stored in the
    * value.
    */
   // TODO: we should rename the method so that it doesn't mention "Context", but "VirtualField"
   void registerMuzzleContextStoreClasses(InstrumentationContextBuilder builder);
+
+  /**
+   * Returns a list of instrumentation helper classes, automatically detected by muzzle during
+   * compilation. Those helpers will be injected into the application classloader.
+   */
+  List<String> getMuzzleHelperClassNames();
+
+  /** See {@link #getMuzzleHelperClassNames()} */
+  static List<String> getMuzzleHelperClassNames(InstrumentationModule module) {
+    if (module instanceof InstrumentationModuleMuzzle) {
+      return ((InstrumentationModuleMuzzle) module).getMuzzleHelperClassNames();
+    } else {
+      return Collections.emptyList();
+    }
```

Comment:
> It is still possible to write custom instrumentations without muzzle

Hold on, if you have to implement `InstrumentationModuleMuzzle` to get helper class names how can you use helper classes without the muzzle plugin now? Add `implements InstrumentationModuleMuzzle` manually to your code?

Reply:
Yes, that is my idea, that if you write custom instrumentation and you don't use muzzle plugin, then you have to apply `InstrumentationModuleMuzzle` interface to your module manually. I will move it back to `extension-api` module.

I still think it is a good idea to have muzzle interface separate from `InstrumentationModule`, even if we have to apply the same stability guarantees to it. To not confuse end-users in the common scenario.

Reply:
> I still think it is a good idea to have muzzle interface separate from `InstrumentationModule`, even if we have to apply the same stability guarantees to it. To not confuse end-users in the common scenario.

💯 
I definitely agree with that.

Reply:
how about `InstrumentationModule.getExtraHelperClassNames()`? it's advanced and therefore potentially confusing to users which is downside, but on the upside it would allow a cleaner separation of muzzle functionality, and we could remove this behavior:

> If your `InstrumentationModule` subclass defines a method with exact same signature as a method
from `InstrumentationModuleMuzzle`, the muzzle compile plugin will not override your code:
muzzle will only generate those methods that do not have a custom implementation.